text/0000-light-dom-scoped-styles.md

```diff
@@ -0,0 +1,233 @@
+---
+title: Light DOM Scoped Styles
+status: DRAFTED
+created_at: 2021-05-14
+updated_at: YYYY-MM-DD
+champion: Nolan Lawson (nolanlawson), Abdulsattar Mohammed (abdulsattar)
+pr: https://github.com/salesforce/lwc-rfcs/pull/50
+---
+
+# Light DOM Scoped Styles
+
+## Summary
+
+This proposal adds a mechanism for light DOM components ([#44](https://github.com/salesforce/lwc-rfcs/pull/44)) to have co-located CSS files that only apply to that component.
+
+## Basic example
+
+Since there is already `*.css` for unscoped light DOM CSS files, we add `*.scoped.css` for scoped light DOM CSS:
+
+```html
+<!-- x/foo.html -->
+<template>
+  <div>Hello</div>
+</template>
+```
+
+```js
+// x/foo.js
+import { LightningElement } from 'lwc'
+
+export default class Foo extends LightningElement {
+  static shadow = false
+}
+```
+
+```css
+/* x/foo.scoped.css */
+div {
+  color: green;
+}
+```
+
+The result will be:
+
+```html
+<x-app>
+  <style class="x-foo_foo" type="text/css">
+    div.x-foo_foo { color: green; }
+  </style>
+  <div class="x-foo_foo">Hello</div>
+</x-app>
+```
+
+## Motivation
+
+For Light DOM ([#44](https://github.com/salesforce/lwc-rfcs/pull/44)), the default assumption is that styles are unscoped. In other words, a light DOM component essentially just contains `<style>` tags that are inserted into the DOM _in situ_. (The actual implementation may be slightly different, but this is the basic mental model.)
+
+Many frameworks, however, have a concept of _scoped styles_, even without using shadow DOM. These offer good developer ergonomics, because developers can concentrate on the CSS co-located with a particular component, without worrying how that CSS might affect other components.
+
+Prior art:
+
+- [Vue: `<style scoped>`](https://vue-loader.vuejs.org/guide/scoped-css.html)
+- [Svelte: scoped styles](https://svelte.dev/docs#style)
+- [React: CSS Modules in `create-react-app`](https://create-react-app.dev/docs/adding-a-css-modules-stylesheet/)
+- [Stencil: scoped styles in light DOM](https://stenciljs.com/docs/styling#scoped-css)
+- [BEM: Block, Element, Modifier](https://en.bem.info/)
+
+## Detailed design
+
+### Scoping token

```

Comment:
I'll add something about this. I agree that in my own mental model, applying the scope token to the root element also felt kind of weird. But it felt weirder _not_ to have it, since otherwise `x-foo` doesn't work.

Reply:
Done, added a section.